presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixConfigurationInitializer.java

```diff
@@ -69,17 +74,20 @@ public RubixConfigurationInitializer(RubixConfig config)
     }
 
     @Override
-    public void initializeConfiguration(Configuration config)
+    public void updateConfiguration(Configuration config, HdfsContext context, URI uri)
     {
+        checkState(getCacheKey(config) == null, "Rubix does not support dynamic FileSystems");
+
         if (!cacheReady) {
             setCacheDataEnabled(config, false);
+            setCacheKey(config, "rubix_disabled");
             return;
         }
 
         updateConfiguration(config);
     }
 
-    public Configuration updateConfiguration(Configuration config)
+    public void updateConfiguration(Configuration config)
```

Comment:
private

Reply:
This needs to stay package-private as it's accessed from `RubixInitializer`